todo_app_bloc/lib/BLoC/todo_bloc.dart

```diff
@@ -0,0 +1,52 @@
+import 'dart:async';
+import 'package:todo_app_bloc/BLoC/todo_event.dart';
+import 'package:uuid/uuid.dart';
+
+class ToDoBloc {
+  var _uuid = Uuid();
+  String _title;
+  String _id;
+
+  ToDoBloc(this._title) {
+    this._id = _uuid.v4().toString();
+  }
+  String get id => _id;
+  String get title => _title;
+
+  @override
+  String toString() => "$title $id";
+}
+
```

Comment:
Appreciate you implementing Bloc using the primitives available in Dart.

Reply:
You could also give a try : https://bloclibrary.dev/#/.
This library has most of the infrastructure required for implements Bloc.